fe/fe-core/src/main/java/org/apache/doris/analysis/ShowPartitionsStmt.java

```diff
@@ -137,8 +151,14 @@ public void analyze(Analyzer analyzer) throws UserException {
     public void analyzeImpl(Analyzer analyzer) throws UserException {
         super.analyze(analyzer);
         tableName.analyze(analyzer);
-        // disallow external catalog
-        Util.prohibitExternalCatalog(tableName.getCtl(), this.getClass().getSimpleName());
+        // disallow unsupported catalog
+        String catalogName = tableName.getCtl();
+        catalog = Env.getCurrentEnv().getCatalogMgr().getCatalog(catalogName);
+        String catalogType = catalog.getType();
+        if (!Strings.isNullOrEmpty(catalogName) && !(catalog.isInternalCatalog() || catalogType.equals("hms"))) {
+            throw new AnalysisException(String.format("Catalog of type '%s' is not allowed in '%s'", catalogType,
+                this.getClass().getSimpleName()));
```

Comment:
Why using `this.getClass().getSimpleName()`?

Reply:
fixed